### PR TITLE
allow single word host names

### LINKED ISSUE #6: Support single word hosts in RequiredHostOptionalPortValidationAdapter
The RequiredHostOptionalPortValidationAdapter doesn't account for single word host names such as "localhost": 

```
[13] pry(#<LogStash::Runner>)> LogStash::PluginMixins::ValidatorSupport::RequiredHostOptionalPortValidationAdapter.validate ["127.0.0.1:3333"]
=> [true, #<struct  host="127.0.0.1", port=3333>]
[14] pry(#<LogStash::Runner>)> LogStash::PluginMixins::ValidatorSupport::RequiredHostOptionalPortValidationAdapter.validate ["localhost:3333"]
=> [false, "Expected a valid required-host optional-port pair, got `\"localhost:3333\"`"]
[15] pry(#<LogStash::Runner>)> LogStash::PluginMixins::ValidatorSupport::RequiredHostOptionalPortValidationAdapter.validate ["localhost.lan:3333"]
=> [true, #<struct  host="localhost.lan", port=3333>]
```

### LINKED ISSUE #15605: Logtash to Logstash native communication output host issue
**Logstash information**:

Logstash 8.11.1 docker image (in docker compose)
docker.elastic.co/logstash/logstash:8.11.1

**Description of the problem**:

Trying the[ Logtash to Logstash native communication](https://www.elastic.co/guide/en/logstash/current/ls-to-ls-native.html)

With a simple output
```json
output {
    stdout {}
    logstash {
        hosts => "logstash:9800"
        ssl_enabled => false
        username => "user"
        password => "secret"
    }
}
```

**Logs**:

```text
2023-11-22 00:39:17 [2023-11-21T23:39:17,065][ERROR][logstash.outputs.logstash] Invalid setting for logstash output plugin:
2023-11-22 00:39:17 
2023-11-22 00:39:17   output {
2023-11-22 00:39:17     logstash {
2023-11-22 00:39:17       # This setting must be a required_host_optional_port
2023-11-22 00:39:17       # Expected a valid required-host optional-port pair, got `"logstash:9800"`
2023-11-22 00:39:17       hosts => "logstash:9800"
2023-11-22 00:39:17       ...
2023-11-22 00:39:17     }
2023-11-22 00:39:17   }
2023-11-22 00:39:17 [2023-11-21T23:39:17,074][ERROR][logstash.agent           ] Failed to execute action {:action=>LogStash::PipelineAction::Create/pipeline_id:main, :exception=>"Java::JavaLang::IllegalStateException", :message=>"Unable to configure plugins: (ConfigurationError) Something is wrong with your configuration.", :backtrace=>["org.logstash.config.ir.CompiledPipeline.<init>(CompiledPipeline.java:120)", "org.logstash.execution.AbstractPipelineExt.initialize(AbstractPipelineExt.java:186)", "org.logstash.execution.AbstractPipelineExt$INVOKER$i$initialize.call(AbstractPipelineExt$INVOKER$i$initialize.gen)", "org.jruby.internal.runtime.methods.JavaMethod$JavaMethodN.call(JavaMethod.java:847)", "org.jruby.ir.runtime.IRRuntimeHelpers.instanceSuper(IRRuntimeHelpers.java:1318)", "org.jruby.ir.instructions.InstanceSuperInstr.interpret(InstanceSuperInstr.java:139)", "org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:367)", "org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:66)", "org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:128)", "org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:115)", "org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:90)", "org.jruby.RubyClass.newInstance(RubyClass.java:931)", "org.jruby.RubyClass$INVOKER$i$newInstance.call(RubyClass$INVOKER$i$newInstance.gen)", "org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:90)", "org.jruby.ir.instructions.CallBase.interpret(CallBase.java:561)", "org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:367)", "org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:66)", "org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:88)", "org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:238)", "org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:225)", "org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:228)", "org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:291)", "org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:328)", "org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:66)", "org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:116)", "org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:136)", "org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:66)", "org.jruby.runtime.IRBlockBody.call(IRBlockBody.java:58)", "org.jruby.runtime.Block.call(Block.java:143)", "org.jruby.RubyProc.call(RubyProc.java:352)", "org.jruby.internal.runtime.RubyRunnable.run(RubyRunnable.java:110)", "java.base/java.lang.Thread.run(Thread.java:840)"]}
```

I don't understand why this configuration is invalid, it's a simple host:port, [like in the doc](https://www.elastic.co/guide/en/logstash/current/ls-to-ls-native.html#configure-upstream-logstash-output), can you help please ?

### PR DESCRIPTION
**Pull request by jsvd**
fixes #6 
fixes https://github.com/elastic/logstash/issues/15605

### CODE REVIEW

File: lib/logstash/plugin_mixins/validator_support/required_host_optional_port_validation_adapter.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ module ValidatorSupport
       is_valid_ipv4 = ->(candidate) { Resolv::IPv4::Regex.match?(candidate) && Resolv::IPv4.create(candidate) rescue false}
       is_valid_ipv6 = ->(candidate) { Resolv::IPv6::Regex.match?(candidate) && Resolv::IPv6.create(candidate) rescue false}
 
-      is_valid_hostname = ->(candidate) { (%r{\A([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}\Z}i).match?(candidate) }
+      is_valid_hostname = ->(candidate) { (%r{\A([a-z0-9]+(-[a-z0-9]+)*\.)?[a-z]{2,}\Z}i).match?(candidate) }
```
Comment (yaauie) [inline lib/logstash/plugin_mixins/validator_support/required_host_optional_port_validation_adapter.rb]: This pattern feels a bit arbitrary, both too lax (`this..that` passes but should fail) and not enough (`asd-f` fails, but should pass).

I wrote and [validated](https://regexr.com/7n2ce) an RFC1123-compliant expression that enforces the format of the labels, the label max lengths, and the whole hostname max length, all without wildcards that could lead to backtracking issues:

```suggestion
      # RFC1123 validated at https://regexr.com/7n2ce
      rfc1123_hostname_pattern = %r{\A(?=.{1,255}\z)(?:(?=[^.]{1,63}(?:[.]|\z))[a-z0-9]+(?:[a-z0-9\-]+[a-z0-9])?(?:[.]|\z))+\z}i
      is_valid_hostname = ->(candidate) { rfc1123_hostname_pattern.match?(candidate) }
```

**Review by yaauie — CHANGES_REQUESTED** [review]
Needs changelog + version bump -> https://github.com/jsvd/logstash-mixin-validator_support/pull/1

**Comment by jsvd** [comment]
@logstashmachine bump patch

**Review by yaauie — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**